connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -462,11 +459,26 @@ private <T> ConfigInfos validateConverterConfig(
         T pluginInstance;
         String stageDescription = "instantiating the connector's " + pluginName + " for validation";
         try (TemporaryStage stage = reportStage.apply(stageDescription)) {
-            pluginInstance = Utils.newInstance(pluginClass, pluginInterface);
+            VersionRange range = PluginVersionUtils.connectorVersionRequirement(pluginVersion);
+            // utils.newInstance is done when no version is provided to preserve older behaviour prior to multi-versioning support
+            // utils.newInstance will try and load the class from the current classloader (which is the connector classloader) before
+            // delegating plugin loading mechanism of the worker, while plugins().newPlugin will always use the worker's plugin loading mechanism
+            // in cases where the converter is packaged with the connector package and no version is provided , utils.newInstance
+            // will choose the converter from the connector, while plugins().newPlugin will choose the latest converter version found while plugin loading.
+            pluginInstance = range == null ? Utils.newInstance(pluginClass, pluginInterface): (T) plugins().newPlugin(pluginClass, range);
```

Comment:
I think this detail is inappropriate in the AbstractHerder, and the classloader choosing logic and etc is more appropriate to handle in the Plugins class.

You can make range == null request the CURRENT_CLASSLOADER for legacy compatibility.

Also i've been trying to eliminate Utils.newInstance in Connect for a while, so don't be afraid of removing it.